### PR TITLE
Adding Proptypes in all components

### PR DESCRIPTION
**Pull request by dersonsena**
As mentioned in issue #38 , basic validations have been added to all components. This minimal validation is **extremely necessary** so that the developer does not forget to place a particular prop one that is required or put of an invalid type.

Proptypes also helps the developer to know which props the component exposes to other components as well as shows the type of each.

As a bonus to the task, I've turned the **Class Component** Task into a **Function Component**. As it is a stateless component, a Function Component makes the code cleaner and easier to understand.

### CODE REVIEW

File: react/src/User.js [hunk 1]
```diff
@@ -89,6 +90,12 @@ class User extends React.Component {
   }
 }
 
+User.propTypes = {
```
Comment (caioflores) [inline react/src/User.js]: When props are not required we need to set a default value don't we? We have this rule in our linter.

Something like this,

`User.defaultProps = {
  user: {}
}`

File: react/src/UserList.js [hunk 2]
```diff
@@ -47,6 +49,11 @@ class UserList extends React.Component {
   }
 }
 
+UserList.propTypes = {
```
Comment (caioflores) [inline react/src/UserList.js]: Same thing here, missing default values.

File: react/src/User.js (hunk 1 above)
Comment (dersonsena) [reply react/src/User.js]: Yes, I'll fix this.

File: react/src/UserList.js (hunk 2 above)
Comment (dersonsena) [reply react/src/UserList.js]: Yes, I'll fix this too.

**Comment by dersonsena** [comment]
@caioflores I added the `defaultProps` in the User and UserList components as suggested.

**Pull request merged**